include/zio/zio_defs.h

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (c) 2019 Kevin Townsend
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ZIO core API definitions.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZIO_DEFS_H_
+#define ZEPHYR_INCLUDE_ZIO_DEFS_H_
+
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ZIO device channel types.
+ *
+ * Each channel kind is associated with a specific SI unit type and scale,
+ * Expressed as the comments below.
+ */
+typedef enum zio_dev_chan_kind {
+	/**
+	 * Raw data channel.
+	 *
+	 * No SI type is prescribed for this channel.
+	 */
+	DEV_CHAN_RAW,
+
+	/**
+	 * Voltage.
+	 *
+	 * Expressed as volts (V).
+	 */
+	DEV_CHAN_VOLTAGE,
+
+	/**
+	 * Current.
+	 *
+	 * Expressed as milli-amps (mA).
+	 */
+	DEV_CHAN_CURRENT,
+
+	/**
+	 * Resistance.
+	 *
+	 * Expressed as ohms.
+	 */
+	DEV_CHAN_RESISTANCE,
+
+	/**
+	 * Capacitance.
+	 *
+	 * Expressed as picofarad (pF).
+	 */
+	DEV_CHAN_CAPACITANCE,
+
+	/**
+	 * Inductance.
+	 *
+	 * Expressed as microhenries (uH).
+	 */
+	DEV_CHAN_INDUCTANCE,
+
+	/**
+	 * Impedance.
+	 *
+	 * Expressed as ohms (Z).
+	 */
+	DEV_CHAN_IMPEDANCE,
+
+	/**
+	 * Frequency.
+	 *
+	 * Expressed as Hertz (Hz).
+	 */
+	DEV_CHAN_FREQUENCY,
+
+	/**
+	 * Generic temperature.
+	 *
+	 * Expressed as degrees celcius (C).
```

Comment:
celsius

Reply:
I always make a mistake on this one, thanks!